### PR TITLE
Pinot Testing cluster

### PR DESCRIPTION
**Pull request by elonazoulay**
Implement pinot testing cluster. Move shared dependency on TestingKafka to presto-testing.

### CODE REVIEW

File: presto-testing/pom.xml [hunk 1]
```diff
@@ -92,6 +92,16 @@
             <artifactId>jackson-annotations</artifactId>
         </dependency>
 
+        <dependency>
```
Comment (kokosing) [inline presto-testing/pom.xml:95]: Maybe you want `presto-testing-kafka`?

File: pom.xml [hunk 2]
```diff
@@ -1114,6 +1114,12 @@
                 <version>3.6.1</version>
```
Comment (kokosing) [inline pom.xml:1114]: > TestingKafka is used by multiple modules for testing.

What modules?

If you are copying it can you please remove testing kafka from other place and reuse your testing kafka?

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/TestingKafka.java [hunk 3]
```diff
@@ -1,92 +0,0 @@
-/*
```
Comment (kokosing) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/TestingKafka.java]: Please squash this commit with the first one.

File: presto-pinot/src/main/java/io/prestosql/pinot/PinotConnectorFactory.java [hunk 4]
```diff
@@ -23,12 +25,20 @@
...
+import java.util.Optional;
 
 import static java.util.Objects.requireNonNull;
 
 public class PinotConnectorFactory
         implements ConnectorFactory
 {
+    private final Optional<Module> extension;
+
+    public PinotConnectorFactory(Optional<Module> extension)
```
Comment (kokosing) [inline presto-pinot/src/main/java/io/prestosql/pinot/PinotConnectorFactory.java:37]: Is this extension needed for host mapper? If not please use separate commit.

File: presto-pinot/src/test/java/io/prestosql/pinot/TestMinimalFunctionality.java [hunk 5]
```diff
@@ -16,31 +16,39 @@
...
+import java.util.Map;
+import java.util.Optional;
 
+import static com.google.inject.multibindings.OptionalBinder.newOptionalBinder;
 import static io.airlift.testing.Closeables.closeAllRuntimeException;
 import static java.util.Objects.requireNonNull;
 
 @Test(singleThreaded = true)
 public class TestMinimalFunctionality
+        extends AbstractTestQueryFramework
```
Comment (kokosing) [inline presto-pinot/src/test/java/io/prestosql/pinot/TestMinimalFunctionality.java:43]: Please squash that commit with `Add TestingPinotCluster`

File: presto-testing/pom.xml (hunk 1 above)
Comment (elonazoulay) [reply presto-testing/pom.xml:95]: Do you mean I should create a separate module?

File: presto-pinot/src/main/java/io/prestosql/pinot/PinotConnectorFactory.java (hunk 4 above)
Comment (elonazoulay) [reply presto-pinot/src/main/java/io/prestosql/pinot/PinotConnectorFactory.java:37]: Yes, it's needed for the hostmapper.

File: pom.xml (hunk 2 above)
Comment (elonazoulay) [reply pom.xml:1114]: That's done in the commit "Use TestingKafka from presto-testing". Should I squash this one with the previous 2 commits?

Comment (kokosing) [reply pom.xml:1114]: Yes, let squash them.

File: presto-testing/pom.xml (hunk 1 above)
Comment (kokosing) [reply presto-testing/pom.xml:95]: Yes

File: presto-testing-kafka/src/main/java/io/prestosql/testing/kafka/TestingKafka.java [hunk 6]
```diff
@@ -34,7 +34,7 @@
 
     public TestingKafka()
     {
-        container = new KafkaContainer("5.4.1");
+        container = new KafkaContainer("5.4.1").withNetworkAliases("kafka");
```
Comment (martint) [inline presto-testing-kafka/src/main/java/io/prestosql/testing/kafka/TestingKafka.java:37]: Unrelated change?

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/TestUtils.java [hunk 7]
```diff
@@ -19,29 +19,20 @@
...
 import java.util.Optional;
-import java.util.Properties;
 
 import static java.lang.String.format;
 
 public final class TestUtils
 {
     private TestUtils() {}
 
-    static Properties toProperties(Map<String, String> map)
```
Comment (martint) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/TestUtils.java:36]: Why did this move?

File: presto-testing-kafka/src/main/java/io/prestosql/testing/kafka/TestingKafka.java (hunk 6 above)
Comment (elonazoulay) [reply presto-testing-kafka/src/main/java/io/prestosql/testing/kafka/TestingKafka.java:37]: I had it in a separate commit but squashed it after previous review.
Should I put it back in a separate commit?

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/TestUtils.java (hunk 7 above)
Comment (elonazoulay) [reply presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/TestUtils.java:36]: This was only used by TestingKafka.

Comment (martint) [reply presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/TestUtils.java:36]: Maybe move it to a separate commit

File: presto-testing-kafka/src/main/java/io/prestosql/testing/kafka/TestingKafka.java (hunk 6 above)
Comment (martint) [reply presto-testing-kafka/src/main/java/io/prestosql/testing/kafka/TestingKafka.java:37]: Yes, since it's a logically independent change, it's better to track it separately.
